### DISCUSSION TITLE
Postgre Run Procedure use prisma client in NextJS

### DISCUSSION THREAD

**Discussion by ainuldentsu** [prisma/prisma#22383] [Q&A]
### Question

Is there any way to run procedure postgresql use prisma client in nextjs? I've read documentation prisma orm section raw queries but I'm still confused how to do that especially I want to run procedure or function postgresql, please give me a reference and sample if any so I can try it.
Thank you.

### Information about Prisma Schema, Client Queries and Environment (optional)

```prisma
// Add your schema.prisma
```

```ts
// Add any relevant Prisma Client queries here
```

- OS: 
- Database: 
- Node.js version: 

- Run `prisma -v` to see your Prisma version and paste it

**Comment by ludralph** [comment]
Hi @ainuldentsu  👋 

Thank you for raising this question.

In order to create a Stored Procedure in Prisma migration, you would need to create a migration without applying it. See [Customize a migration](https://www.prisma.io/docs/guides/migrate/developing-with-prisma-migrate/customizing-migrations), this will allow you edit a migration file and add the neccessary SQL stored procedure function. Take a look at the previously answered [discussion](https://github.com/prisma/prisma/discussions/20865) for how to implement a stored procedure in Prisma.

If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Reply by ainuldentsu** [reply]
Hi @ludralph,

Thanks for you explanation, after I read it, still not match with expect me, the procedure that you mention earlier to cover migration, but my problem is how to run procedure that the content is insert two table sequentially with second insert use id from first table, this for the procedure:
`CREATE OR REPLACE PROCEDURE do_insert_two_tabl(param1 varchar(50), param2 varchar(50))
LANGUAGE plpgsql
AS $$
BEGIN

INSERT INTO table1
("field_t1")
VALUES(param1);

SELECT field1 from table1 t1 order by t1.created_at desc;

INSERT INTO table1
("field_t2", "field_t2_2")
VALUES(param2, field1);

END
$$;`

if I run in sql editor like this:
call return_params('Client');

then what way that I can run same procedure use prisma client in nextjs?
Hopefully can deepen and clarify my question,
Thank you.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
